ios/Classes/RudderSdkFlutterPlugin.m

```diff
@@ -200,18 +195,18 @@ -(RSOption*) getRudderOptionsObject:(NSDictionary *) optionsDict {
     RSOption * options = [[RSOption alloc]init];
     if([optionsDict objectForKey:@"externalIds"])
     {
-      NSArray *externalIdsArray =  [optionsDict objectForKey:@"externalIds"];
-      for(NSDictionary *externalId in externalIdsArray) {
-        [options putExternalId:[externalId objectForKey:@"type"] withId:[externalId objectForKey:@"id"]];
-       } 
+        NSArray *externalIdsArray =  [optionsDict objectForKey:@"externalIds"];
+        for(NSDictionary *externalId in externalIdsArray) {
+            [options putExternalId:[externalId objectForKey:@"type"] withId:[externalId objectForKey:@"id"]];
+        }
     }
     if([optionsDict objectForKey:@"integrations"])
     {
-      NSDictionary *integrationsDict = [optionsDict objectForKey:@"integrations"];
-      for(NSString* key in integrationsDict)
-      {
-          [options putIntegration:key isEnabled:[[integrationsDict objectForKey:key] boolValue]];
-      }
+        NSDictionary *integrationsDict = [optionsDict objectForKey:@"integrations"];
+        for(NSString* key in integrationsDict)
+        {
+            [options putIntegration:key isEnabled:[[integrationsDict objectForKey:key] isEqual: @1] ? YES : NO];
```

Comment:
Can you please explain why this would be necessary?

Reply:
isEnabled parameter is Expecting a type of`BOOL` and if we try to send the value by casting using `boolValue` as before it is considering the value as true if it is not null. So even if it's false it is being considered true. So we are directly sending a `BOOL` to overcome this behaviour